src/board/system76/common/flash/wrapper.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2020 Evan Lojewski
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+#include <board/flash.h>
+#include <common/ec.h>
+
+#if __EC__ == it5570e
+// IT5502: SCAR0 is stored in processor cache, not in xram
+volatile uint8_t __xdata __at(0x1040) SCAR0L;
+volatile uint8_t __xdata __at(0x1041) SCAR0M;
+volatile uint8_t __xdata __at(0x1042) SCAR0H;
+
+#define SCARL SCAR0L
+#define SCARM SCAR0M
+#define SCARH SCAR0H
+
+#elif __EC__ == it8587e
+// IT8587: SCAR1 is in xram at 0x800-0xC00
+volatile uint8_t __xdata __at(0x1043) SCAR1L;
+volatile uint8_t __xdata __at(0x1044) SCAR1M;
+volatile uint8_t __xdata __at(0x1045) SCAR1H;
+
+#define SCARL SCAR1L
+#define SCARM SCAR1M
+#define SCARH SCAR1H
+
+#else
+#error Unknown EC
+#endif
+
+
+void flash_entry(uint32_t addr, uint8_t * data, uint8_t length, uint8_t command) __reentrant;
```

Comment:
If you have a .h file shared between flash ROM and the main ROM, and included it in flash/main.c, then this definition would be checked by the compiler to match the definition in flash/main.c